duckduckgo/iOS
# Fix sync connect flow

Task/Issue URL: https://app.asana.com/0/0/1204398270301335/f
Tech Design URL:
CC:

**Description**:

Fix connect flow on iOS.   Also some minor restructuring and error handling.

**Steps to test this PR**:
1. On macOS turn off sync, then turn on sync, choose sync another device 
2. On iOS turn off sync, then turn on sync, choose sync another device and scan the code (or copy/paste it from macOS if that's easier)
3. The UI will close and the other device will connect.  Eventually the new device will appear in the UI.




**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
